cmd/agent/app/reporter/grpc/builder_test.go

```diff
@@ -116,16 +115,18 @@ func TestBuilderWithCollectors(t *testing.T) {
 			cfg.Discoverer = test.discoverer
 
 			conn, err := cfg.CreateConnection(zap.NewNop())
-			if err != nil {
-				assert.Equal(t, test.err, err)
-			} else {
-				assert.NotNil(t, conn)
+			if test.expectedError == "" {
+				require.NoError(t, err)
+				require.NotNil(t, conn)
 
 				if test.checkSuffixOnly {
 					assert.True(t, strings.HasSuffix(conn.Target(), test.target))
 				} else {
 					assert.True(t, conn.Target() == test.target)
 				}
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedError)
```

Comment:
https://godoc.org/github.com/stretchr/testify/assert#EqualError is more concise

Reply:
That function performs exact comparison of the error message. Since the message is wrapped multiple times, I don't know or care what the underlying stdlib error was, I only care about the substring introduced by the code under test.